### PR TITLE
removed incorrect stuff from docs for v6 release. added discovery examples

### PR DESCRIPTION
**Pull request by woutslakhorst**
This is some minor cleanup. It doesn't yet cover what is potentially missing.

### CODE REVIEW

File: docs/pages/deployment/discovery.rst [hunk 1]
```diff
@@ -68,4 +81,70 @@ The endpoint for a Discovery Service MUST be in the following form (unless mappe
...
+                  }
+                }
+              ]
+            }
+          }
+        ]
+      }
+    }
+
+A service definition consists of:
```
Comment (gerardsn) [inline docs/pages/deployment/discovery.rst:145]: link RFC (that is not published)

File: docs/pages/deployment/key-rotation.rst [hunk 2]
```diff
@@ -30,17 +27,6 @@ Then, you add a new key which generates a new key pair in your crypto storage an
...
-
-2. Remove a key
-===============
-
-To remove a key from the DID document.
-
-.. code-block:: shell
-
-    DELETE /internal/vdr/v2/did/{did}/verificationmethod/{kid}
+    POST /internal/vdr/v2/did/{subject}/verificationmethod
```
Comment (gerardsn) [inline docs/pages/deployment/key-rotation.rst:30]:
```suggestion
    POST /internal/vdr/v2/subject/{id}/verificationmethod
```

File: docs/pages/deployment/key-rotation.rst [hunk 3]
```diff
@@ -30,17 +27,6 @@ Then, you add a new key which generates a new key pair in your crypto storage an
...
 
-When successful, it returns with a ``204`` tatus code
+When successful, it returns the verification method(s) that were added to the DID document.
```
Comment (gerardsn) [inline docs/pages/deployment/key-rotation.rst:32]:
```suggestion
When successful, it returns the verification method(s) that were added to the DID document(s).
```

File: docs/pages/deployment/storage.rst [hunk 4]
```diff
@@ -60,7 +60,6 @@ Microsoft Azure Key Vault
 This storage backend uses Microsoft Azure's Key Vault. It authenticates to the Azure Key Vault at the configured URL using the default credential,
 typically an Azure Managed Identity. Refer to the `Azure SDK for Go documentation <https://github.com/Azure/azure-sdk-for-go/wiki/Set-up-Your-Environment-for-Authentication>`_ for more information.
 
-- Azure Key Vault storage can't be used for nodes that produce ``did:nuts`` DIDs or for data encryption.
```
Comment (gerardsn) [inline docs/pages/deployment/storage.rst:63]: I think this is still true?

File: docs/pages/deployment/discovery.rst [hunk 5]
```diff
@@ -35,17 +35,30 @@ If multiple query parameters are specified, all of them must match a single Veri
 Registration
 ============
 
-To register a DID on a Discovery Service, the DID must be activated for the service.
-The Nuts node will then register a Verifiable Presentation of the DID on the service, and periodically refresh it.
-E.g., for service ``coffeecorner`` and DID ``did:web:example.com``:
+To register a subject on a Discovery Service, the subject must be activated for the service.
+The Nuts node will then register a Verifiable Presentation for all subject DIDs on the service, and periodically refresh it.
+E.g., for service ``coffeecorner`` and subject ``example`` (``did:web:example.com``, ``did:nuts:ec6d1834-11e0-48a4-b3f6-934268c7b870``):
```
Comment (reinkrul) [inline docs/pages/deployment/discovery.rst:40]: The DIDs only add confusipn here?

File: docs/pages/deployment/discovery.rst [hunk 6]
```diff
@@ -35,17 +35,30 @@ If multiple query parameters are specified, all of them must match a single Veri
...
 
-    POST /internal/discovery/v1/coffeecorner/did:web:example.com
+    POST /internal/discovery/v1/coffeecorner/example
 
 The DID's wallet must contain the Verifiable Credential(s) that are required by the service definition,
 otherwise registration will fail. If the wallet does not contain the credentials,
-the Nuts node will retry registration periodically.
+the Nuts node will retry registration periodically for all DIDs of a subject.
+
+Optionally, a POST body can be sent which contains `RegistrationParameters`:
```
Comment (reinkrul) [inline docs/pages/deployment/discovery.rst:50]: Sent -> provided

File: docs/pages/deployment/discovery.rst [hunk 7]
```diff
@@ -35,17 +35,30 @@ If multiple query parameters are specified, all of them must match a single Veri
...
+
+.. code-block:: json
+
+    {
+      "registrationParameters": {
+        "endpoint": "https://api.example.com",
+        "contact": "alice@example.com"
+    }
+
```
Comment (reinkrul) [inline docs/pages/deployment/discovery.rst:59]: I think it needs an additionele newline after the code block, Github shows it in red

File: docs/pages/integrating/vc.rst [hunk 8]
```diff
@@ -68,10 +70,22 @@ Will be expanded by the node to:
       ]
     }
 
-The `visibility` property indicates the contents of the VC are published on the network, so it can be read by everyone.
+There are three parameters that can be passed:
+- `format` (optional): The format of the VC. Can be ``ldp_vc`` or ``jwt_vc``. Default is ``ldp_vc``.
+- `publishToNetwork` (optional): Whether the VC should be published on the network. Default is ``true``.
+- `visibility` (optional): The visibility of the VC. Can be ``public`` or ``private``. Default is ``private``.
```
Comment (reinkrul) [inline docs/pages/integrating/vc.rst:76]: These are for credentials issued from did:Nuts only

**Review by reinkrul — CHANGES_REQUESTED** [review]

File: docs/pages/integrating/vc.rst [hunk 9]
```diff
@@ -68,10 +70,22 @@ Will be expanded by the node to:
...
+- `withStatusList2021Revocation` (optional): Whether the VC should be issued with a status list 2021 revocation. Default is ``false``.
 
-By default, the node will create credentials in JSON-LD format.
-You can specify the format by passing the `format` parameter (``jwt_vc`` or ``ldp_vc``).
+These defaults are set based on the V5 defaults. This works for the did:nuts method but might not work for other methods.
+The preferred way in V6 is to pass the following parameters:
+- `publishToNetwork`: ``false``
+- `withStatusList2021Revocation`: ``true``
```
Comment (gerardsn) [inline docs/pages/integrating/vc.rst:82]: statuslist should only be used for long lived credentials. It's preferred to choose the expiration time such that revocation is not necessary.

**Review by gerardsn — CHANGES_REQUESTED** [review]

File: docs/pages/deployment/discovery.rst (hunk 7 above)
Comment (woutslakhorst) [reply docs/pages/deployment/discovery.rst:59]: no a `}`

File: docs/pages/deployment/storage.rst (hunk 4 above)
Comment (woutslakhorst) [reply docs/pages/deployment/storage.rst:63]: no there's a SQL table in between now

File: docs/pages/integrating/vc.rst (hunk 9 above)
Comment (woutslakhorst) [reply docs/pages/integrating/vc.rst:82]: removed entire section

File: docs/pages/deployment/storage.rst (hunk 4 above)
Comment (gerardsn) [reply docs/pages/deployment/storage.rst:63]: but that doesn't change that did:nuts uses ecies to encrypt the `pal` header that is not supported by azure.
Or can the header also be encrypted using RSA?

Comment (woutslakhorst) [reply docs/pages/deployment/storage.rst:63]: ahh yes, you can't use did:nuts with private credentials over the network

File: docs/pages/deployment/key-rotation.rst (hunk 2 above)
Comment (gerardsn) [reply docs/pages/deployment/key-rotation.rst]: this is not fixed

Comment (woutslakhorst) [reply docs/pages/deployment/key-rotation.rst]: it was pretty close...

**Review by gerardsn — APPROVED** [review]

**Review by reinkrul — APPROVED** [review]

**Pull request merged**